E0130 needs to be updated to new format

From: src/test/compile-fail/E0130.rs

Error E0130 needs a span_label, updating it from:

```
error[E0130]: patterns aren't allowed in foreign function declarations
  --> src/test/compile-fail/E0130.rs:12:12
   |
12 |     fn foo((a, b): (u32, u32)); //~ ERROR E0130
   |            ^^^^^^
```

To:

```
error[E0130]: patterns aren't allowed in foreign function declarations
  --> src/test/compile-fail/E0130.rs:12:12
   |
12 |     fn foo((a, b): (u32, u32)); //~ ERROR E0130
   |            ^^^^^^ pattern not allowed in foreign function
```

--- a/src/librustc_passes/ast_validation.rs
+++ b/src/librustc_passes/ast_validation.rs
@@ -183,6 +183,7 @@ impl<'a> Visitor for AstValidator<'a> {
                                                    E0130,
                                                    "patterns aren't allowed in foreign function \
                                                     declarations");
+                    err.span_label(span, &format!("pattern not allowed in foreign function"));
                     if is_recent {
                         err.span_note(span,
                                       "this is a recent error, see issue #35203 for more details");

--- a/src/test/compile-fail/E0130.rs
+++ b/src/test/compile-fail/E0130.rs
@@ -9,7 +9,9 @@
 // except according to those terms.
 
 extern {
-    fn foo((a, b): (u32, u32)); //~ ERROR E0130
+    fn foo((a, b): (u32, u32));
+    //~^ ERROR E0130
+    //~| NOTE pattern not allowed in foreign function
 }
 
 fn main() {

--- a/src/test/compile-fail/no-patterns-in-args.rs
+++ b/src/test/compile-fail/no-patterns-in-args.rs
@@ -10,10 +10,13 @@
 
 extern {
     fn f1(mut arg: u8); //~ ERROR patterns aren't allowed in foreign function declarations
-                        //~^ NOTE this is a recent error
+                        //~^ NOTE pattern not allowed in foreign function
+                        //~| NOTE this is a recent error
     fn f2(&arg: u8); //~ ERROR patterns aren't allowed in foreign function declarations
+                     //~^ NOTE pattern not allowed in foreign function
     fn f3(arg @ _: u8); //~ ERROR patterns aren't allowed in foreign function declarations
-                        //~^ NOTE this is a recent error
+                        //~^ NOTE pattern not allowed in foreign function
+                        //~| NOTE this is a recent error
     fn g1(arg: u8); // OK
     fn g2(_: u8); // OK
     // fn g3(u8); // Not yet

